go/tools/gopackagesdriver/bazel_json_builder.go

```diff
@@ -17,72 +17,104 @@ package main
 import (
 	"context"
 	"fmt"
+	"path/filepath"
 	"strings"
 )
 
 type BazelJSONBuilder struct {
-	bazel      *Bazel
-	query      string
-	tagFilters string
-	targets    []string
+	bazel    *Bazel
+	requests []string
 }
 
 const (
 	OutputGroupDriverJSONFile = "go_pkg_driver_json_file"
 	OutputGroupStdLibJSONFile = "go_pkg_driver_stdlib_json_file"
 	OutputGroupExportFile     = "go_pkg_driver_export_file"
+	wildcardQuery             = `kind("go_library", //...)`
 )
 
-func NewBazelJSONBuilder(bazel *Bazel, query, tagFilters string, targets []string) (*BazelJSONBuilder, error) {
+func (b *BazelJSONBuilder) fileQuery(filename string) string {
+	if filepath.IsAbs(filename) {
+		fp, _ := filepath.Rel(b.bazel.WorkspaceRoot(), filename)
+		filename = fp
+	}
+	return fmt.Sprintf(`some(kind("go_library", same_pkg_direct_rdeps("%s")))`, filename)
+}
+
+func (b *BazelJSONBuilder) packageQuery(importPath string) string {
+	return fmt.Sprintf(`some(kind("go_library", attr(importpath, "%s", //...)))`, importPath)
```

Comment:
Queries like this would take a very long time to finish in a repository with 100K targets and 2000 third-party libraries, like Uber's Go monorepo

Reply:
several things we can do:
1. have the scope configurable via environment
2. restrict the scope to the file's package, which is most of the time where the file lives
3. for `importpath` queries, this is trickier as the package could be anywhere in `//...`, however, package queries don't happen when using `gopls`, as it only gives out `file=` ones

Reply:
actually `file=` queries use `same_pkg_direct_rdeps` so it shouldn't take long when editing

Reply:
ah wait! it's the universe scope of `//...` that loads package anyway. let me try something else.